py/torch_tensorrt/dynamo/conversion/impl/pool.py

```diff
@@ -169,3 +172,451 @@ def end_index(idx: int, out_dim: int, in_dim: int) -> int:
 
     output = impl.cat.cat(ctx, target, source_ir, f"{name}_cat", output_list, dim=-1)
     return output
+
+
+def adaptive_avg_pool2d(
+    ctx: ConversionContext,
+    target: Union[Target, str],
+    source_ir: Optional[SourceIR],
+    name: str,
+    input: TRTTensor,
+    output_size: Sequence[int],
+) -> TRTTensor:
+    input_shape = input.shape
+    input_rank = len(input_shape)
+    need_reshape_back = False
+
+    if input_rank == 3:  # reshape to 4D for TRT pooling
+        input = impl.shuffle.reshape(
+            ctx, target, source_ir, f"{name}_reshape", input, (1, *input.shape)
+        )
+        need_reshape_back = True
+        input_shape = input.shape
+        input_rank = len(input_shape)
+
+    extend_len = len(output_size)
+    output_size = list(output_size)
+    original_input = input
+
+    # repeat_interleave the input if the dim of output is larger than input
+    insert_axises = []
+    for axis in range(1, extend_len + 1):
+        axis = -axis
+        positive_axis = get_positive_dim(
+            axis, input_rank
+        )  # convert to positive axis, which is for calculating new shapes below
```

Comment:
Is this the same as `axis % input_rank`? Asked since the negation + re-cast to positive could be confusing here

Reply:
Sorry for the confusion. This is because I want to reversely traverse the input. For example, the tuple is [0,1,2,3], for the first iteration, `axis = 1; axis = -axis = -1; positive_axis = get_positive_dim(-1, 4) = 3`. Both `axis` and `positive_axis` are recorded for the following use.